tests/test_storm_control.py

```diff
@@ -0,0 +1,528 @@
+        dvs.start_swss()
+        time.sleep(10)
+
+        for key in tbl.getKeys():
+            self.check_storm_control_on_interface(dvs,key,"broadcast",1000000)
+            self.check_storm_control_on_interface(dvs,key,"unknown-unicast",2000000)
+            self.check_storm_control_on_interface(dvs,key,"unknown-multicast",3000000)
+
+    def test_add_storm_lag_interface(self,dvs,testlog):
+        self.setup_db(dvs)
+        lag_name = "PortChannel10"
+        member_interface = "Ethernet0"
+        kbps_value = 1000000
+        storm_list = ["broadcast","unknown-unicast","unknown-multicast"]
+        kbps_value_list = [1000000,2000000,3000000]
+
+        #Create VLAN interface and add member
+        self.create_port_channel(dvs,lag_name)
+        self.add_port_channel_member(dvs,lag_name,member_interface)
+
+        #click CLI verification
+        for storm_type in storm_list:
+            dvs.runcmd("config interface storm-control "+storm_type+" add "+lag_name+" "+str(kbps_value))
+            tbl = swsscommon.Table(self.cdb,"PORT_STORM_CONTROL")
+            (status,fvs) = tbl.get(lag_name+"|"+storm_type)
+            assert status == False
+            assert len(fvs) == 0
+
+        #Orchagent verification
+        storm_list_db = ["broadcast","unknown-unicast","unknown-multicast"]
+        for storm_type,kbps_value in zip(storm_list_db,kbps_value_list):
+            #Cleanup syslog
+            dvs.runcmd(['sh', '-c', "echo 0 > /var/log/syslog"])
+            time.sleep(1)
+            print ("storm type: "+storm_type+" kbps value: "+str(kbps_value))
+            #Add storm entry to config DB directly
+            self.add_storm_session(lag_name,storm_type,kbps_value)
+            tbl = swsscommon.Table(self.cdb,"PORT_STORM_CONTROL")
+            (status,fvs) = tbl.get(lag_name+"|"+storm_type)
+            print ("config DB table content ")
+            print (fvs)
+            assert status == True
+            assert len(fvs) > 0
+            time.sleep(1)
+            #grep for error message in syslog
+            (exitcode,num) = dvs.runcmd(['sh', '-c', 'cat /var/log/syslog | grep -i "handlePortStormControlTable: {}: Unsupported / Invalid interface PortChannel10"'.format(storm_type)])
+            time.sleep(1)
+            assert exitcode == 0
+
+    def test_add_storm_vlan_interface(self,dvs,testlog):
+        self.setup_db(dvs)
+        vlan_id = 10
+        member_interface = "Ethernet4"
+        kbps_value = 1000000
+        storm_list = ["broadcast","unknown-unicast","unknown-multicast"]
+        kbps_value_list = [1000000,2000000,3000000]
+        vlan_name = "Vlan"+str(vlan_id)
+
+        #Create VLAN interface and add member
+        self.create_vlan(dvs,str(vlan_id))
+        self.add_vlan_member(dvs,str(vlan_id),member_interface)
+
+        #click CLI verification 
+        for storm_type in storm_list:
+            dvs.runcmd("config interface storm-control "+storm_type+" add Vlan"+str(vlan_id)+" "+str(kbps_value))
```

Comment:
Why isn't validation not done at the CLI level? do not allow config on Vlan/PortChannel interface

Reply:
Validation is done in CLI level already. This was written to validate Orchagent behavior.